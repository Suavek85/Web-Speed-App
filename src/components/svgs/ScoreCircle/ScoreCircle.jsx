import React from "react"
import PropTypes from 'prop-types'
import classNames from 'classnames/bind'

import styles from './ScoreCircle.scss'

export default function ScoreCircle(props) {

  const { score } = props
  const scoreCircle = score ? Math.floor(score * 100) : 0
  const strokeValue = `${ scoreCircle }, 100`

  const isAlert = !!(scoreCircle < 50)
  const isWarning = !!(scoreCircle < 90 && scoreCircle > 49)
  const isInfo = !!(scoreCircle > 89)

  //STYLES
  const cx = classNames.bind(styles)

  const getClass = cx({
    blockStats: true,
    blockStatsAlert: isAlert,
    blockStatsWarning: isWarning,
    blockStatsInfo: isInfo
  })

  const getTextClass = cx({
    blockText: true,
    blockTextAlert: isAlert,
    blockTextWarning: isWarning,
    blockTextInfo: isInfo
  })

  const getBgClass = cx({
    blockBg: true,
    blockBgAlert: isAlert,
    blockBgWarning: isWarning,
    blockBgInfo: isInfo
  })

  return (
    <div className={ styles.block }>
      <svg viewBox="0 0 36 36" className={ styles.blockSvg }>
        <path className={ getBgClass }
          d="M18 2.0845
            a 15.9155 15.9155 0 0 1 0 31.831
            a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <path className={ getClass }
          strokeDasharray={ strokeValue }
          d="M18 2.0845
            a 15.9155 15.9155 0 0 1 0 31.831
            a 15.9155 15.9155 0 0 1 0 -31.831"
        />
        <text x="18" y="21.5" className={ getTextClass }>{ scoreCircle }</text>
      </svg>
    </div>
  
  )
}

ScoreCircle.propTypes = {
  score: PropTypes.string,
}
