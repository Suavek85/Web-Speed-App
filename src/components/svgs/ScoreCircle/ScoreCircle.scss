@import '../../../scss/styles';

$block: chart;
//svg
$blockSvg: elem($block, svg);
//stats
$blockStats: elem($block, stats);
$blockStatsAlert: modifier($blockStats, alert);
$blockStatsWarning: modifier($blockStats, warning);
$blockStatsInfo: modifier($blockStats, info);
//text
$blockText: elem($block, text);
$blockTextAlert: modifier($blockText, alert);
$blockTextWarning: modifier($blockText, warning);
$blockTextInfo: modifier($blockText, info);
//background
$blockBg: elem($block, background);
$blockBgAlert: modifier($blockBg, alert);
$blockBgWarning: modifier($blockBg, warning);
$blockBgInfo: modifier($blockBg, info);

.#{$block} {
  width: 48px;
  height: 48px;
  margin-left: 7px;

  &__svg {
    display: block;
  }

  &__background {
    fill: #fff0d8;
    stroke: #ffda9d;
    stroke-width: 3.8;

    &--alert {
      fill: #ffd8d8;
      stroke: #ff8989;
    }

    &--warning {
      fill: #fff0d8;
      stroke: #ffda9d;
    }

    &--info {
      fill: #ceffce;
      stroke: #80ff80;
    }
  }

  &__stats {
    fill: none;
    stroke-linecap: round;
    stroke-width: 3.6;

    &--alert {
      stroke: #f00;
    }

    &--warning {
      stroke: #ff9f00;
    }

    &--info {
      stroke: $color-green-success;
    }
  }

  &__text {
    font-family: sans-serif;
    font-size: em(12, 13);
    font-weight: 500;
    text-anchor: middle;

    &--alert {
      fill: #f00;
    }

    &--warning {
      fill: #ff9f00;
    }

    &--info {
      fill: $color-green-success;
    }
  }
}

:export {
  block: $block;
  blockSvg: $blockSvg;
  blockBg: $blockBg;
  blockStats: $blockStats;
  blockStatsAlert: $blockStatsAlert;
  blockStatsWarning: $blockStatsWarning;
  blockStatsInfo: $blockStatsInfo;
  blockText: $blockText;
  blockTextAlert: $blockTextAlert;
  blockTextWarning: $blockTextWarning;
  blockTextInfo: $blockTextInfo;
  blockBgAlert: $blockBgAlert;
  blockBgWarning: $blockBgWarning;
  blockBgInfo: $blockBgInfo;
}
