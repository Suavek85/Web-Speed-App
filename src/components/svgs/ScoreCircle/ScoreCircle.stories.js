import React from 'react'
import { storiesOf } from '@storybook/react'
import { withKnobs } from "@storybook/addon-knobs"
import ScoreCircle from './ScoreCircle'

const stories = storiesOf('SVGs', module)
stories.addDecorator(withKnobs)

stories.add('Score Circle svg', () => {
  return <ScoreCircle score='0.40' />
})