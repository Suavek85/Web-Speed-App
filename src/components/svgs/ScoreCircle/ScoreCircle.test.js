import React from 'react'
import * as redux from 'react-redux'
import { shallow } from 'enzyme'
import ScoreCircle from './ScoreCircle'

describe('ScoreCircle snapshot', () => {

  it('should match the snapshot', () => {
    const spyUseSelector = jest.spyOn(redux, 'useSelector')
    spyUseSelector.mockReturnValue('INACTIVE')
    const wrapper = shallow(<ScoreCircle />)
    expect(wrapper).toMatchSnapshot()
  })

})