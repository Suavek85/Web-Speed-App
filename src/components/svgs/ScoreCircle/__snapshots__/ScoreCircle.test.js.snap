// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ScoreCircle snapshot should match the snapshot 1`] = `
<div>
  <svg
    viewBox="0 0 36 36"
  >
    <path
      className="blockBg blockBgAlert"
      d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
    />
    <ForwardRef
      className="blockStats blockStatsAlert"
      d="M18 2.0845 a 15.9155 15.9155 0 0 1 0 31.831 a 15.9155 15.9155 0 0 1 0 -31.831"
      style={
        Object {
          "strokeDasharray": SpringValue {
            "_children": Set {
              [Function],
            },
            "_defaultProps": Object {},
            "_lastAsyncId": 0,
            "_phase": "CREATED",
            "_priority": 0,
            "_state": Object {
              "key": "strokeDasharray",
            },
            "_timestamps": Object {},
            "animation": Object {
              "value": [Circular],
            },
            "id": 2,
            "key": "strokeDasharray",
            "node": AnimatedString {
              "_string": null,
              "_toString": [Function],
              "_value": 0,
              "done": true,
              "lastPosition": 0,
            },
          },
        }
      }
    />
    <ForwardRef
      className="blockText blockTextAlert"
      x="18"
      y="21.5"
    >
      0
    </ForwardRef>
  </svg>
</div>
`;
